source/features/mute-users.tsx

```diff
@@ -0,0 +1,134 @@
+		return;
+	}
+
+	select('.minimized-comment', comment)!.classList.add('d-none');
+	select('.minimized-comment .Details-element', comment)!.setAttribute('open', 'true');
+	select('.unminimized-comment', comment)!.classList.remove('d-none');
+}
+
+async function onMuteUnmuteClick(event: React.MouseEvent<HTMLButtonElement>): Promise<void> {
+	let mutedUsers = await getMutedUsers();
+
+	const comment = (event.target as HTMLElement).closest('.js-comment-container')!;
+	const viewportOffset = comment.getBoundingClientRect().top;
+	const user = select('.author', comment)!.textContent!;
+
+	if (mutedUsers.includes(user)) {
+		mutedUsers = mutedUsers.filter(mutedUser => mutedUser !== user);
+	} else {
+		mutedUsers.push(user);
+	}
+
+	await setMutedUsers(mutedUsers);
+
+	const avatars = select.all(`.js-discussion .js-comment-container .avatar[alt="@${user}"`);
+	const comments = avatars.map(avatar => avatar.closest('.js-comment-container') as HTMLElement);
+
+	for (const comment of comments) {
+		if (mutedUsers.includes(user)) {
+			minimizeComment(comment);
+		} else {
+			unminimizeComment(comment);
+		}
+	}
+
+	requestAnimationFrame(() => {
+		const newOffset = comment.getBoundingClientRect().top;
+		window.scrollBy(0, newOffset - viewportOffset);
+	});
+}
+
+async function handleMenuOpening(event: DelegateEvent): Promise<void> {
+	const dropdown = select('.show-more-popover', event.delegateTarget.parentElement!)!;
+	const user = select('.author', dropdown.closest('.js-comment-container')!)!.textContent!;
+
+	if (user === currentUser) {
+		return;
+	}
+
+	const mutedUsers = await getMutedUsers();
+	const isMutedUser = mutedUsers.includes(user);
+
+	const existingButton = select('.rgh-mute-unmute-button', event.delegateTarget.parentElement!);
+	if (existingButton) {
+		if (isMutedUser) {
+			existingButton.textContent = 'Unmute user';
+		} else {
+			existingButton.textContent = 'Mute user';
+		}
+
+		return;
+	}
+
+	// Add option to mute or unmute user
+	appendBefore(dropdown, 'a[data-ga-click^="Report"]',
+		<button
+			className="dropdown-item btn-link rgh-mute-unmute-button"
+			role="menuitem"
+			title={`${isMutedUser ? 'Unmute' : 'Mute'} this user`}
+			onClick={onMuteUnmuteClick}>
+			{isMutedUser ? 'Unmute' : 'Mute'} user
+		</button>
+	);
+}
+
+async function minimizeMutedUserComments(): Promise<void> {
+	const mutedUsers = await getMutedUsers();
+
+	for (const comment of select.all('.js-discussion .js-comment-container')) {
+		const user = select('.author', comment)!.textContent!;
+
+		// If the user _is_ muted, minimize their comment
+		if (mutedUsers.includes(user) && user !== currentUser) {
+			minimizeComment(comment);
+		}
+	}
+}
+
+function init(): void {
+	currentUser = getUsername();
+	minimizeMutedUserComments();
+
+	delegate('.timeline-comment-action', 'click', handleMenuOpening);
+}
+
+features.add({
+	id: 'mute-users',
+	description: 'Mute users to minimize comments from them',
```

Comment:
This might need improvement, but I can’t think of anything more obvious than just “mute users”

Reply:
Agree, because "mute" means to me that I'll not get any notifications from them anymore, which is not the case.

Maybe "**ignore**". It doesn't sound as bad as "blocking".

Reply:
> This might need improvement, but I can’t think of anything more obvious than just “mute users”

"Mute users" seems obvious but that doesn't tell us what the feature actually does.

Reply:
Yeah that’s why I was asking for a better one since I couldn’t think of a better description. 

@jerone Perhaps the feature should be `minimize-user` or something like that? Mute is self-explanatory but it indeed it doesn’t work like the general Mute feature found on socials.

Reply:
What about `minimize-a-users-comments`

Reply:
`minimize-user` sounds like the right description here. 👍

Reply:
👍 

“Ignore” is used by GitHub the same way “Mute” is, actually. See the repo “Watch” dropdown

Reply:
Yes, `minimize-user-comments` sounds fine. But this means we'd have to change the label used in user dropdowns to something other than "Mute/Unmute user".

Maybe "Minimize user comments" and "_Maximize_ user comments"?

Reply:
> minimize-user-comments

👍 

> Maybe "Minimize user comments" and "Maximize user comments"?

- `Minimize user's comment`
- `Maximize user's comment`